sympy/physics/hydrogen.py

```diff
@@ -132,11 +132,11 @@ def Psi_nlm(n, l, m, r, phi, theta, Z=1):
     # sympify arguments
     n, l, m, r, phi, theta, Z = map(S, [n, l, m, r, phi, theta, Z])
     # check if values for n,l,m make physically sense
-    if n.is_integer and n < 1:
+    if n.is_integer == True and n.is_constant() and n < 1:
         raise ValueError("'n' must be positive integer")
```

Comment:
Maybe you could do this like
```
    if n.is_integer is False or (n < 1) is True:
        raise ValueError
```

Reply:
@oscarbenjamin, Thank you for the review.

Reply:
Why is `is_constant` needed?

Reply:
That's because if n is a symbol with `integer=True`, we wouldn't be able to compare it to 1 as it would raise an error.

Reply:
It's not `n < 1` that raises it's `bool(n < 1)`. That's why I suggested using `(n < 1) is True` above.

Reply:
Ohh I understand it now. Sorry, I didn't get that at first. I have made the necessary changes.